src/CMakeLists.txt

```diff
@@ -58,6 +58,21 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
 endif(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
 set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS}")
 
+include(CheckCXXSourceCompiles)
+CHECK_CXX_SOURCE_COMPILES("
+int main() {
+#ifdef __GLIBCXX__
+#if __GLIBCXX__ < 20150422ul /* libstdc++ < 5.1.0 */
+#error slow std::list::size()
+#endif
+#endif
+}
+" HAVE_GLIBCXX_5_1)
+if(NOT HAVE_GLIBCXX_5_1)
+  # libc++'s list::size() is O(1) also.
+  message(WARNING "performance regression is expected due to an O(n) implementation of 'std::list::size()' in libstdc++ older than 5.1.0")
```

Comment:
should this change to ERROR then?

Reply:
then, we need to install ubuntu-toolchain-r on trusty.

Reply:
Wait, Mimic will still support trusty?

Reply:
Hmm, I'm not sure we've discussed it.  Is trusty the only one that is problematic here?  I would very happily drop trusty!

Reply:
b/c xenial comes with gcc 5.3, and centos7 will be using devtoolset-7, (hope @ktdreyer is fine with this). and current debian stable (stretch) offers gcc 6.3. the debian's old stable (jessie) offers gcc 4.9. SLE 11 has gcc-6.

but i don't think serious user of centos/rhel 7 and debian jessie would ignore the performance regression warning and continue using whatever the toolchain the distro offers.

so the only blocker is ubuntu trusty. and it's fixable, we just need to update the pbuilderrc script and set the `DEB_CC` and `DEB_CXX` env variables. i am still looking into this.

but if we can drop the trusty, that'd be easier.

Reply:
Let's drop trusty.

Would you please not use devtoolset-7 until we can discuss it directly with the team at Red Hat that maintains that? There is no published EOL date for that, so it could become unsupported at any time, and their security problems become ours.

The last time I talked with the SCL team they were very hesitant to publish clear EOL dates on centos.org. We need to bring this up with them again.

Reply:
> SLE 11 has gcc-6

@tchaikov I guess you meant SLE 12 (the current version). SLE 11 is very old and no recent Ceph runs on it. I checked and the current SLE-12-SP3 has gcc 6.2.1.

Reply:
thanks for confirming @smithfarm! no, i did mean SLE 11. b/c i found that SLE-11.4's EOL is [2019-03-31](https://en.wikipedia.org/wiki/SUSE_Linux#SUSE_distributions), that's why i thought we should consider SLE 11.4. and i was wrong, it has gcc-5.2 not gcc-6, per the [release notes](https://www.suse.com/releasenotes/x86_64/SUSE-SLES/11-SP4/#fate-319015)

anyway, it's great that we don't need to worry about SLE =)

Reply:
@ktdreyer sorry for using devtoolset-7 without any notice beforehand. i thought it's the future of devtoolset and will be well supported. may i know which devtoolset is supported? so we can use it?